src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -550,21 +515,25 @@ final case class ClientApiServiceImpl(
 
   }
 
-  private def checkAuthorizationForRoles(roles: String, relationshipIds: String, requesterId: UUID, userId: UUID)(
-    implicit contexts: Seq[(String, String)]
-  ): Future[List[UUID]] = {
-    if (roles.contains(SECURITY_ROLE)) getRelationship(userId, requesterId, Seq(SECURITY_ROLE)).map(List[UUID](_))
-    else parseArrayParameters(relationshipIds).traverse(_.toFutureUUID)
+  private def checkAuthorizationForRoles(
+    roles: String,
+    userIds: String,
+    selfcareUuid: UUID,
+    requesterId: UUID,
+    userId: UUID
+  )(implicit contexts: Seq[(String, String)]): Future[List[UUID]] = {
+    if (roles.contains(SECURITY_ROLE)) getUser(selfcareUuid, requesterId, userId, Seq(SECURITY_ROLE)).map(List[UUID](_))
```

Comment:
here we already have the userId required for the filter
```suggestion
    if (roles.contains(SECURITY_ROLE)) userId :: Nil
```